tests/end-to-end/opa-gatekeeper-policies.bats

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bats
+
+# Tests OPA opa gatekeeper policies -
+
+setup_file() {
+  load "../common/lib"
+  load "../common/lib/harbor"
+
+  common_setup
+
+  harbor.load_env gatekeeper-policies
+  harbor.setup_project
+  harbor.setup_user_demo_image
+
+  export harbor_endpoint
+
+  export user_demo_chart="${ROOT}/tests/common/docs/user-demo/deploy/ck8s-user-demo"
+  export user_demo_image
+
+  if ! docker pull "${user_demo_image}"; then
+    fail "unable to pull image from harbor"
+  fi
+
+  if [[ "$(yq wc '.user.namespaces[] | select(. == "staging")')" != "staging" ]]; then
+    fail "these tests requires that you have a 'staging' user namespace"
+  fi
```

Comment:
Should this perhaps be checked via kubectl? Since you could have added the staging namespace via subnamespace.

Reply:
The check has been reworked.